addons/website_form/controllers/main.py

```diff
@@ -32,6 +32,14 @@ def website_form(self, model_name, **kwargs):
             # I couldn't find a cleaner way to pass data to an exception
             return json.dumps({'error_fields' : e.args[0]})
 
+        geoip_country_code = request.session.geoip and request.session.geoip.get('country_code') or False
+        if geoip_country_code:
+            country_id = request.env['res.country'].sudo().search([('code', '=', geoip_country_code)], limit=1)
```

Comment:
don't use Sudo without reason: "access_res_country_group_all","res_country group_user_all","model_res_country",,1,0,0,0
It is public !

Sudo will recompute a new env at each time, so it is a useless cost.

Limit 1 is useless: ('code_uniq', 'unique (code)', 'The code of the country must be unique !')